packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -179,9 +193,30 @@ export async function runHttpQuery(
         }
       }
 
-      let context = optionsObject.context || {};
-      if (typeof context === 'function') {
-        context = context();
+      let context = optionsObject.context;
+      if (!context) {
+        //appease typescript compiler, otherwise could use || {}
+        context = {};
+      } else if (typeof context === 'function') {
+        try {
+          context = await context();
```

Comment:
So isn't this a backwards-incompatible change?  Doesn't this break non-async context functions?

I mean that's OK because this is 2.0 I guess, but let's not bury the lede here...

Reply:
await just unwraps a Promise if it's there. If the value on the rhs of `await` is an object or value, then it is returned immediately. I think this is bug fix or at minimum a strict addition to api to allow the context to return a Promise in addition to a value. What this change does break is context functions that returns a promise and the resolvers expect the context argument to be a Promise. I think this case is pretty unlikely and not a good best practice. 

https://github.com/apollographql/apollo-server/pull/1024/files/4225c08b10c90dad5c0acc0b27fc72f10482ad12#diff-5102db46bcc7f6cc0c59149e9cc375d3R26 is the type addition to the context

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await
shows an example that has `await 20`.